pandas/util/testing.py

```diff
@@ -1185,8 +1201,13 @@ def assert_series_equal(
                 right._internal_get_values(),
                 check_dtype=check_dtype,
             )
-    elif is_interval_dtype(left) or is_interval_dtype(right):
-        assert_interval_array_equal(left.array, right.array)
+    elif is_interval_dtype(left) or is_interval_dtype(left):
+        # must cast to interval dtype to keep mypy happy
```

Comment:
What was the complaint on this? This changes the actual assertions being done I think

Reply:
If I don't cast to IntervalArray I get these errors 
```
pandas/util/testing.py:1211: error: Argument 1 to "assert_interval_array_equal" has incompatible type "ExtensionArray"; expected "IntervalArray"
pandas/util/testing.py:1211: error: Argument 2 to "assert_interval_array_equal" has incompatible type "ExtensionArray"; expected "IntervalArray"
```

Reply:
I've removed the asserts though, they weren't needed here.

Reply:
As mentioned by @jreback should use `cast` from the typing module here - don't want to actually construct a new object via a call to `IntervalArray`